### PR TITLE
pass request to get client

### PR DESCRIPTION
**Pull request by chainlink**
Happy to update docs, or extend this to all model methods :)

### CODE REVIEW

**Comment by chainlink** [comment]
Related https://github.com/oauthjs/node-oauth2-server/issues/254

File: lib/grant-types/password-grant-type.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ PasswordGrantType.prototype.getUser = function(request) {
     throw new InvalidRequestError('Invalid parameter: `password`');
   }
 
-  return promisify(this.model.getUser, 2)(request.body.username, request.body.password)
+  return promisify(this.model.getUser, 2)(request.body.username, request.body.password, request)
```
Comment (maxtruxa) [inline lib/grant-types/password-grant-type.js]: The second argument to `promisify` reflects the number of arguments the function takes.

```
return promisify(this.model.getUser, 3)(request.body.username, request.body.password, request)
//                                   ^
```

**Comment by chainlink** [comment]
Fixed!

**Comment by mjsalinger** [comment]
See comments on #336 to refactor this PR and also please extend it to all model methods for completeness.

**Comment by mjsalinger** [comment]
Hi @chainlink, are you going to be updating this PR or should we close?

**Comment by chainlink** [comment]
Hey, sorry I haven't had time to extend this beyond client :( Close for now I guess?

**Comment by mjsalinger** [comment]
Closing due to lack of activity.

**Comment by NicolaiSchmid** [comment]
Hey, where exactly would changes be needed? 
Are we talking about the other methods (eg getClient, saveToken etc) inside the password-grant or about all methods in all grant-types?

**Pull request closed without merging**